src/imitation/scripts/ingredients/demonstrations.py

```diff
@@ -109,43 +141,15 @@ def generate_expert_trajs(
         )
 
 
-@demonstrations_ingredient.capture
-def load_local_expert_trajs(
-    rollout_path: Union[str, pathlib.Path],
-    n_expert_demos: Optional[int],
-) -> Sequence[types.Trajectory]:
-    """Loads expert demonstrations from a local path.
-
-    Args:
-        rollout_path: A path containing a pickled sequence of `types.Trajectory`.
-        n_expert_demos: The number of trajectories to load.
-            Dataset is truncated to this length if specified.
-
-    Returns:
-        The expert trajectories.
-
-    Raises:
-        ValueError: There are fewer trajectories than `n_expert_demos`.
-    """
-    expert_trajs = serialize.load(rollout_path)
-    logger.info(f"Loaded {len(expert_trajs)} expert trajectories from '{rollout_path}'")
-    if n_expert_demos is not None:
-        if len(expert_trajs) < n_expert_demos:
-            raise ValueError(
-                f"Want to use n_expert_demos={n_expert_demos} trajectories, but only "
-                f"{len(expert_trajs)} are available via {rollout_path}.",
-            )
-        expert_trajs = expert_trajs[:n_expert_demos]
-        logger.info(f"Truncated to {n_expert_demos} expert trajectories")
-    return expert_trajs
-
-
 @demonstrations_ingredient.capture(prefix="expert")
-def _download_expert_rollouts(rollout_type, loader_kwargs):
+def _download_expert_rollouts(rollout_type: str, loader_kwargs: Dict[str, Any]):
     assert rollout_type.endswith("-huggingface")
     algo_name = rollout_type.split("-")[0]
-    return serialize.load_rollouts_from_huggingface(
+    model_name = hfsb3.ModelName(
         algo_name,
-        env_name=loader_kwargs["env_name"],
-        organization=loader_kwargs["organization"],
+        hfsb3.EnvironmentName(loader_kwargs["env_name"]),
     )
+    repo_id = hfsb3.ModelRepoId(loader_kwargs["organization"], model_name)
```

Comment:
While we are making this change we may want to add a bit of future proofing. First we might want to be able to have datasets that are more fine-grained than just algorithm + environment. For example, we might want a dataset created from a PPO agent trained on ant for 100,000 time steps and another dataset for the same agent trained after 1,000,000 time steps. In addition, we might want to update the dataset lets say we find a bug in the PPO code and upload a new dataset to the hub. For experiment repeatability people should be able to select the old `revision` of the dataset.

Reply:
You are totally right. I looked again into the API design of this and cleaned it up a bit:

Now you can either specify `demonstrations.organization` (default [`HumanCompatibleAI`](https://huggingface.co/HumanCompatibleAI)) and `demonstrations.algo_name` (default `ppo`). Then the HuggingFace Hub `repo_id` is deduced according to `<organization>/<algo_name>-<normalized_env_name>` (e.g. [`HumanCompatibleAI/ppo-seals-Walker2d-v0`](https://huggingface.co/HumanCompatibleAI/ppo-seals-Walker2d-v0)) or you specify `demonstrations.path`. Then this is used as the `repo_id`.

The `loader_kwargs` are now part of the `demonstrations` ingredient and not pulled from the `expert` ingredient. They are passed straight to the `datasets.load_dataset()` function so you have full control over how the dataset is loaded. You can even select your own split here (the default value for the `loader_kwargs` is `{"split": "train"}`).

I also changed the design so you have to explicitly set the `demonstrations.type` to either `local`, `huggingface` or `generated`. This fixes two issues:
1. before we awkwardly extracted the `algo_name` from the `type` which was set to `<algo_name>-huggingface`.
2. We implicitly started to generate demonstrations if the `path` was not set. This could lead to unexpected behavior and was a bit harder to debug.